library/std/src/panicking.rs

```diff
@@ -233,6 +235,42 @@ where
     *hook = Hook::Custom(Box::new(move |info| hook_fn(&prev, info)));
 }
 
+/// Adjusts the visual representation of panic messages by enabling or
+/// disabling color highlighting, or leaving it with a default behaviour. This
+/// function sets, unsets or removes an environment variable (RUST_COLOR_ERRORS)
+/// accordingly, influencing downstream formatting functions to produce colorful
+/// error messages.
+#[unstable(feature = "panic_color_errors", issue = "none")]
+pub fn highlight_errors(set_color: Option<bool>){
+    match set_color {
+        Some(true) => {crate::env::set_var("RUST_COLOR_ERRORS", "1")}
+        Some(false) => {crate::env::set_var("RUST_COLOR_ERRORS", "0")}
+        _ => {crate::env::remove_var("RUST_COLOR_ERRORS")}
+    }
+}
+
+/// Alters the format of error messages by adding color codes to them.
+/// It reads the RUST_COLOR_ERRORS environment variable to determine whether to
+/// apply color formatting. If enabled, it formats the error message with ANSI
+/// escape codes to display it in red, indicating an error state. The default
+/// behavior is to apply highlighting when printing to a terminal, and no
+/// highlighting otherwise.
+#[unstable(feature = "panic_color_errors", issue = "none")]
+fn format_error_message (msg: &str) -> String {
+    match crate::env::var_os("RUST_COLOR_ERRORS") {
```

Comment:
We should cache this instead of checking it every time, as we do with most env vars.